packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -41,8 +41,17 @@ type Connection<NodesType> =
     };
 
 type PaginationState<NodesType> = {
-  nodes?: Array<NodesType>;
-  pageInfo?: PageInfo | null;
+  pagination?: {
+    [key: string]: {
+      nodes: Array<NodesType>;
+      pageInfo: {
+        endCursor: Maybe<string> | undefined;
+        startCursor: Maybe<string> | undefined;
+        hasPreviousPage: boolean;
+        hasNextPage: boolean;
+      };
```

Comment:
What was the reason for inlining this rather than keeping the `PageInfo` type here?